### PR TITLE
Add OCI Privileged container support

### PR DESCRIPTION
**Pull request by jterry75**
Signed-off-by: Justin Terry (VM) <juterry@microsoft.com>

### CODE REVIEW

File: service/gcs/core/gcs/devices.go [hunk 1]
```diff
@@ -0,0 +1,105 @@
+package gcs
+
+// TODO: JTERRY75 this is from runc/devices.go. When we fix the vendor issue pull this in from that pacakge instead.
```
Comment (jhowardmsft) [inline service/gcs/core/gcs/devices.go]: 👍

File: service/gcs/core/gcs/uvm.go [hunk 2]
```diff
@@ -169,6 +169,55 @@ func (h *Host) CreateContainer(id string, settings *prot.VMHostedContainerSettin
...
+	// Check if we need to do any capability/device mappings
+	if settings.OCISpecification.Linux.Resources != nil {
+		if settings.OCISpecification.Linux.Resources.Devices != nil {
+			// We use the device list access/allow to detect the privileged
+			// case. Check that here.
+			if len(settings.OCISpecification.Linux.Resources.Devices) == 1 {
+				if settings.OCISpecification.Linux.Resources.Devices[0].Allow == true ||
+					settings.OCISpecification.Linux.Resources.Devices[0].Access == "rwm" {
+
+					logrus.Debugf("opengcs::CreateContainer - spec.Linux.Resources.Devices set for privileged container")
```
Comment (jhowardmsft) [inline service/gcs/core/gcs/uvm.go]: Not sure the logging protocol throughout this area of code, but should this be `WithFields`...?

File: service/gcs/core/gcs/devices.go [hunk 3]
```diff
@@ -0,0 +1,105 @@
...
+	ErrNotADevice = errors.New("not a device node")
+)
+
+// Testing dependencies
+var (
+	unixLstat     = unix.Lstat
+	ioutilReadDir = ioutil.ReadDir
+)
+
+// Given the path to a device and its cgroup_permissions(which cannot be easily queried) look up the information about a linux device and return that information as a Device struct.
```
Comment (kevpar) [inline service/gcs/core/gcs/devices.go]: nit: line-wrap comment

Comment (jterry75) [reply service/gcs/core/gcs/devices.go]: This isnt our code. Leaving it alone since I will be removing.

Comment (kevpar) [reply service/gcs/core/gcs/devices.go]: I missed this was copied code on the first pass.

**Pull request merged**